ACE/ace/CDR_Base.cpp

```diff
@@ -948,8 +951,8 @@ void ACE_CDR::Fixed::normalize (UShort min_scale)
 
 ACE_CDR::Fixed ACE_CDR::Fixed::from_string (const char *str)
 {
-  const bool negative = str && *str == '-';
-  if (negative || (str && *str == '+'))
+  const bool negative = *str == '-';
```

Comment:
Coverity is still flagging it as an issue. If there's a way to specify a known precondition, could you do that, else fix the code to check null str first and bail before doing content checks?